server/java/src/main/java/com/godaddy/asherah/grpc/AppEncryptionServer.java

```diff
@@ -0,0 +1,207 @@
+  private static String productId;
+  @CommandLine.Option(names = "--serviceId", description = "Specify the service id", required = true)
+  private static String serviceId;
+  @CommandLine.Option(names = "--uds", description = "Unix domain socket file",
+      defaultValue = "/tmp/appencryption.sock", required = true)
+  private static String udsFilePath;
+
+  // Options to set up crypto policy
+  @CommandLine.Option(names = "--key-expiration-days", defaultValue = "90",
+      description = "The number of days after which a key will expire")
+  private static int keyExpirationDays;
+  @CommandLine.Option(names = "--revoke-check-minutes", defaultValue = "60",
+      description = "Sets the cache's TTL to revoke the keys in the cache")
+  private static int revokeCheckMinutes;
+  @CommandLine.Option(names = "--session-caching", defaultValue = "false",
+      description = "Enable/disable the session caching")
+  private static boolean sessionCacheEnabled;
+  @CommandLine.Option(names = "--session-cache-max-size", defaultValue = "1000",
+      description = "Define the number of maximum sessions to cache.")
+  private static int sessionCacheMaxSize;
+  @CommandLine.Option(names = "--session-cache-expire-minutes", defaultValue = "2",
+      description = "Evict the session from cache after some minutes.")
+  private static int sessionCacheExpireMinutes;
+
+  private static SessionFactory sessionFactory;
+
+  private AppEncryptionServer() {
+  }
+
+  private NettyServerBuilder bindUDS(final String uds) {
+
+    // Create a new server to listen on a socket
+    EventLoopGroup group;
+    NettyServerBuilder builder = NettyServerBuilder.forAddress(new DomainSocketAddress(uds));
+    if (SystemUtils.IS_OS_MAC) {
+      group = new KQueueEventLoopGroup();
+      builder.channelType(KQueueServerDomainSocketChannel.class);
+    }
+    else if (SystemUtils.IS_OS_LINUX) {
+      group = new EpollEventLoopGroup();
+      builder.channelType(EpollServerDomainSocketChannel.class);
+    }
+    else {
+      throw new IllegalStateException("binding to unix:// addresses is only supported on Linux and macOS");
+    }
+    builder.workerEventLoopGroup(group).bossEventLoopGroup(group);
+
+    return builder;
+  }
+
+  private Metastore<JSONObject> setupMetastore() {
+
+    if (metastoreType == MetastoreType.JDBC) {
+      if (jdbcUrl != null) {
+        logger.info("using JDBC-based metastore...");
+
+        // Setup JDBC persistence from command line argument using Hikari connection pooling
+        HikariDataSource dataSource = new HikariDataSource();
+        dataSource.setJdbcUrl(jdbcUrl);
+        return JdbcMetastoreImpl.newBuilder(dataSource).build();
+      }
+      else {
+        CommandLine.usage(this, System.out);
+        return null;
+      }
+    }
+    else if (metastoreType == MetastoreType.DYNAMODB) {
+      logger.info("using DynamoDB-based metastore...");
+
+      return DynamoDbMetastoreImpl.newBuilder().build();
+    }
```

Comment:
Can be simplified. Consider using a `switch` or reduce nesting (see inverted if example below).